Mathlib/Analysis/CStarAlgebra/CompletelyPositiveMap.lean

```diff
@@ -0,0 +1,148 @@
+/-
+Copyright (c) 2025 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import Mathlib.Analysis.CStarAlgebra.PositiveLinearMap
+import Mathlib.Analysis.CStarAlgebra.CStarMatrix
+
+/-! # Completely positive maps
+
+A linear map `φ : A₁ →ₗ[ℂ] A₂` (where `A₁` and `A₂` are C⋆-algebras) is called
+*completely positive (CP)* if `CStarMatrix.map (Fin k) (Fin k) φ` (i.e. applying `φ` to all
+entries of a k × k matrix) is also positive for every `k ∈ ℕ`.
+
+This file defines completely positive maps and develops their basic API.
+
+## Main results
+
++ `NonUnitalStarAlgHomClass.instCompletelyPositiveMapClass`: Non-unital star algebra
+  homomorphisms are completely positive.
+
+## Notation
+
++ `A₁ →CP A₂` denotes the type of CP maps from `A₁` to `A₂`.
+
+## Implementation notes
+
+The morphism class `CompletelyPositiveMapClass` is designed to be part of the order hierarchy,
+and only includes the order property; linearity is not mentioned at all. It is therefore meant
+to be used in conjunction with `LinearMapClass`. This is meant to avoid mixing order and algebra
+as much as possible.
+-/
+
+/--
+A linear map `φ : A₁ →ₗ[ℂ] A₂`  is called *completely positive (CP)* if
+`CStarMatrix.mapₗ (Fin k) (Fin k) φ` (i.e. applying `φ` to all entries of a k × k matrix) is also
+positive for every `k ∈ ℕ`.
+-/
+structure CompletelyPositiveMap (A₁ : Type*) (A₂ : Type*) [NonUnitalCStarAlgebra A₁]
+    [NonUnitalCStarAlgebra A₂] [PartialOrder A₁] [PartialOrder A₂] [StarOrderedRing A₁]
+    [StarOrderedRing A₂] extends A₁ →ₗ[ℂ] A₂ where
+  map_cstarMatrix_nonneg' (k : ℕ) (M : CStarMatrix (Fin k) (Fin k) A₁) (hM : 0 ≤ M) :
+      0 ≤ M.map toLinearMap
+
+/--
+A linear map `φ : A₁ →ₗ[ℂ] A₂`  is called *completely positive (CP)* if
+`CStarMatrix.mapₗ (Fin k) (Fin k) φ` (i.e. applying `φ` to all entries of a k × k matrix) is also
+positive for every `k ∈ ℕ`.
+-/
+class CompletelyPositiveMapClass (F : Type*) (A₁ : Type*) (A₂ : Type*)
+    [NonUnitalCStarAlgebra A₁] [NonUnitalCStarAlgebra A₂] [PartialOrder A₁]
+    [PartialOrder A₂] [StarOrderedRing A₁] [StarOrderedRing A₂] [FunLike F A₁ A₂] where
+  map_cstarMatrix_nonneg' (φ : F) (k : ℕ) (M : CStarMatrix (Fin k) (Fin k) A₁) (hM : 0 ≤ M) :
+    0 ≤ M.map φ
```

Comment:
At first I was confused about why you restrict to `Fin k` here and not arbitrary `[Fintype n]`. However, I think I understand that the point is including a generic `n` would introduce another universe parameter. Correct? If so, please explain in the docstring and link to the lemma that does it for generic `n`.

Same comment applies to `CompletelyPositiveMap`.

Reply:
Yes that's right. I added the explanation in the two docstrings.